dartagnan/src/main/java/com/dat3m/dartagnan/program/Events.java

```diff
@@ -0,0 +1,355 @@
+package com.dat3m.dartagnan.program;
+
+import com.dat3m.dartagnan.expression.*;
+import com.dat3m.dartagnan.expression.op.BOpUn;
+import com.dat3m.dartagnan.expression.op.COpBin;
+import com.dat3m.dartagnan.expression.op.IOpBin;
+import com.dat3m.dartagnan.program.arch.aarch64.event.StoreExclusive;
+import com.dat3m.dartagnan.program.arch.linux.event.*;
+import com.dat3m.dartagnan.program.arch.linux.event.cond.*;
+import com.dat3m.dartagnan.program.arch.tso.event.Xchg;
+import com.dat3m.dartagnan.program.atomic.event.*;
+import com.dat3m.dartagnan.program.event.*;
+import com.dat3m.dartagnan.program.event.pthread.*;
+import com.dat3m.dartagnan.program.event.rmw.*;
+import com.dat3m.dartagnan.program.memory.Address;
+import com.dat3m.dartagnan.program.svcomp.event.BeginAtomic;
+import com.dat3m.dartagnan.program.svcomp.event.EndAtomic;
+
+import java.math.BigInteger;
+
+public class Events {
```

Comment:
I think EventFactory would be a better name. It is more conventional and less confusing with the `Event` class

Reply:
I thought about it. But I think it gets unecessarily long and quite redundant if we use ```Factory``` on every layer, e.g.
```EventFactory.PthreadFactory``` is redundant. It makes sense if we use singleton classes but not for purely static classes.
Also the top-level class needs to have the same name as the file AFAIK, so ```ProgramFactory``` as a file name is not possible if the top level class is called ```EventFactory```. Either way, ```ProgramFactory``` is IMO not a good name cause the factory doesn't produce programs but only events.

In short, I can rename the file and the top-level class to ```EventFactory``` but I would keep the inner classes as is.